tree.go

```diff
@@ -1405,8 +1381,8 @@ func (n *LeafNode) GetProofItems(keys keylist) (*ProofElements, []byte, [][]byte
 // Serialize serializes a LeafNode.
 // The format is: <nodeType><stem><bitlist><c1comm><c2comm><children...>
 func (n *LeafNode) Serialize() ([]byte, error) {
-	cBytes := banderwagon.ElementsToBytes([]*banderwagon.Element{n.c1, n.c2})
-	return n.serializeWithCompressedCommitments(cBytes[0], cBytes[1]), nil
+	cBytes := banderwagon.ElementsToBytes([]*banderwagon.Element{n.commitment, n.c1, n.c2})
+	return n.serializeLeafWithCompressedCommitments(cBytes[0], cBytes[1], cBytes[2]), nil
```

Comment:
You mentioned in the PR description that:
> .. so that the uncompressed form can be stored inside the child's node itself

Did you meant "compressed" form? Or should we change the code? Or maybe is deferred?
(Uncompressed is attractive to avoid costly deserialization later, which I think was part of the idea (apart from saving space))

Reply:
the goal is to have the uncompressed form, but right now I just want to see the feasibility of path-based, to compare the performance - then if we can improve by not serializing points, that's even better - but let's see.